pkg/kubelet/cm/cgroup_manager_linux.go

```diff
@@ -591,3 +594,102 @@ func (m *cgroupManagerImpl) GetResourceStats(name CgroupName) (*ResourceStats, e
 	}
 	return toResourceStats(stats), nil
 }
+
+func readCgroupFile(dir, file string) (string, error) {
+	data, errRead := ioutil.ReadFile(filepath.Join(dir, file))
+	if errRead != nil {
+		return "", fmt.Errorf("failed to read cgroup file %s in path %s: %v", file, dir, errRead)
+	}
+	return string(data), nil
+}
+
+func readInt64CgroupFile(dir, file string) (int64, error) {
+	data, err := readCgroupFile(dir, file)
+	if err != nil {
+		return 0, err
+	}
+	return strconv.ParseInt(strings.TrimSpace(data), 10, 64)
+}
+
+func readUint64CgroupFile(dir, file string) (uint64, error) {
+	data, err := readCgroupFile(dir, file)
+	if err != nil {
+		return 0, err
+	}
+	return strconv.ParseUint(strings.TrimSpace(data), 10, 64)
+}
+
+// Get the memory limit in bytes applied to the cgroup
+func (m *cgroupManagerImpl) GetCgroupMemoryLimit(name CgroupName) (uint64, error) {
+	cgroupPaths := m.buildCgroupPaths(name)
+	stats, err := getStatsSupportedSubsystems(cgroupPaths)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get stats supported cgroup subsystems for cgroup %v: %v", name, err)
+	}
+	return stats.MemoryStats.Usage.Limit, nil
+}
+
+// Get the cpu quota, cpu period, and cpu shares applied to the cgroup
+func (m *cgroupManagerImpl) GetCgroupCpuLimit(name CgroupName) (int64, uint64, uint64, error) {
+	cgroupPaths := m.buildCgroupPaths(name)
+	cgroupCpuPath, found := cgroupPaths["cpu"]
+	if !found {
+		return 0, 0, 0, fmt.Errorf("failed to build CPU cgroup fs path for cgroup %v", name)
+	}
+	cpuQuota, errQ := readInt64CgroupFile(cgroupCpuPath, "cpu.cfs_quota_us")
+	if errQ != nil {
+		return 0, 0, 0, fmt.Errorf("failed to read CPU quota for cgroup %v: %v", name, errQ)
+	}
+	cpuPeriod, errP := readUint64CgroupFile(cgroupCpuPath, "cpu.cfs_period_us")
+	if errP != nil {
+		return 0, 0, 0, fmt.Errorf("failed to read CPU period for cgroup %v: %v", name, errP)
+	}
+	cpuShares, errS := readUint64CgroupFile(cgroupCpuPath, "cpu.shares")
+	if errP != nil {
+		return 0, 0, 0, fmt.Errorf("failed to read CPU shares for cgroup %v: %v", name, errS)
+	}
+	return cpuQuota, cpuPeriod, cpuShares, nil
+}
+
+// Set the memory limit in bytes applied to the cgroup
+func (m *cgroupManagerImpl) SetCgroupMemoryLimit(name CgroupName, memoryLimit int64) error {
+	cgroupPaths := m.buildCgroupPaths(name)
+	cgroupMemoryPath, found := cgroupPaths["memory"]
+	if !found {
+		return fmt.Errorf("failed to build memory cgroup fs path for cgroup %v", name)
+	}
+	memLimit := strconv.FormatInt(memoryLimit, 10)
```

Comment:
why 10?

Reply:
It is for decimal conversion. This is consistent with rest of the k8s codebase, and really with other code as well. See this from latest k8s upstream branch:
`root@fw0000358:~/MASTER_K8S# git grep "strconv\.FormatInt" 
cmd/kubeadm/app/apis/kubeadm/apiendpoint.go:    return net.JoinHostPort(endpoint.AdvertiseAddress, strconv.FormatInt(int64(endpoint.BindPort), 10))
cmd/kubeadm/app/util/config/initconfiguration.go:                       cfg.ControlPlaneEndpoint = net.JoinHostPort(host, strconv.FormatInt(int64(LocalAPIEndpoint.BindPort), 10))
pkg/api/v1/resource/helpers.go: return strconv.FormatInt(c, 10), nil
pkg/api/v1/resource/helpers.go: return strconv.FormatInt(m, 10), nil
pkg/api/v1/resource/helpers.go: return strconv.FormatInt(m, 10), nil
pkg/apis/apps/v1/conversion.go:         out.Annotations[appsv1.DeprecatedRollbackTo] = strconv.FormatInt(in.Spec.RollbackTo.Revision, 10)
pkg/apis/apps/v1/conversion.go: out.Annotations[appsv1.DeprecatedTemplateGeneration] = strconv.FormatInt(in.Spec.TemplateGeneration, 10)
pkg/apis/apps/v1beta2/conversion.go:            out.Annotations[appsv1beta2.DeprecatedRollbackTo] = strconv.FormatInt(in.Spec.RollbackTo.Revision, 10)
pkg/apis/apps/v1beta2/conversion.go:    out.Annotations[appsv1beta2.DeprecatedTemplateGeneration] = strconv.FormatInt(in.Spec.TemplateGeneration, 10)
pkg/controller/daemon/daemon_controller_test.go:        obj.SetResourceVersion(strconv.FormatInt(ver+1, 10))
pkg/controller/deployment/deployment_controller_test.go:        obj.SetResourceVersion(strconv.FormatInt(ver+1, 10))
pkg/controller/deployment/rollback.go:  d.Annotations[apps.DeprecatedRollbackTo] = strconv.FormatInt(rollbackTo.Revision, 10)
pkg/controller/deployment/sync.go:      newRevision := strconv.FormatInt(maxOldRevision+1, 10)
pkg/controller/deployment/util/deployment_util_test.go: return types.UID(strconv.FormatInt(rand.Int63(), 10))
`

-----

`root@fw0000358:~/MASTER_K8S# git grep "strconv\.FormatInt" vendor/
vendor/github.com/Azure/azure-sdk-for-go/storage/blockblob.go:          headers["Content-Length"] = strconv.FormatInt(n, 10)
vendor/github.com/Azure/azure-sdk-for-go/storage/util.go:                                       val = strconv.FormatInt(reflectedValue.Int(), 10)
vendor/github.com/BurntSushi/toml/encode.go:            enc.wf(strconv.FormatInt(rv.Int(), 10))
vendor/github.com/aws/aws-sdk-go/aws/credentials/stscreds/web_identity_provider.go:             sessionName = strconv.FormatInt(now().UnixNano(), 10)
vendor/github.com/aws/aws-sdk-go/aws/csm/metric.go:     return []byte(strconv.FormatInt(int64(ns/time.Millisecond), 10)), nil
vendor/github.com/aws/aws-sdk-go/aws/ec2metadata/api.go:        ttl := strconv.FormatInt(int64(duration/time.Second), 10)
vendor/github.com/aws/aws-sdk-go/aws/signer/v4/v4.go:           ctx.Query.Set("X-Amz-Expires", strconv.FormatInt(duration, 10))
vendor/github.com/aws/aws-sdk-go/private/protocol/query/queryutil/queryutil.go:         v.Set(name, strconv.FormatInt(value, 10))
vendor/github.com/aws/aws-sdk-go/private/protocol/rest/build.go:                str = strconv.FormatInt(value, 10)
vendor/github.com/aws/aws-sdk-go/private/protocol/timestamp.go:         return strconv.FormatInt(t.Unix(), 10)
vendor/github.com/aws/aws-sdk-go/private/protocol/xml/xmlutil/build.go:         str = strconv.FormatInt(converted, 10)
vendor/github.com/cespare/prettybench/prettybench.go:   return strconv.FormatInt(int64(iter), 10)
vendor/github.com/cilium/ebpf/asm/alu_string.go:                return "Source(" + strconv.FormatInt(int64(i), 10) + ")"
vendor/github.com/cilium/ebpf/asm/alu_string.go:                return "Endianness(" + strconv.FormatInt(int64(i), 10) + ")"
vendor/github.com/cilium/ebpf/asm/alu_string.go:        return "ALUOp(" + strconv.FormatInt(int64(i), 10) + ")"
vendor/github.com/cilium/ebpf/asm/func_string.go:               return "BuiltinFunc(" + strconv.FormatInt(int64(i), 10) + ")"
vendor/github.com/cilium/ebpf/asm/jump_string.go:       return "JumpOp(" + strconv.FormatInt(int64(i), 10) + ")"
vendor/github.com/cilium/ebpf/asm/load_store_string.go:         return "Mode(" + strconv.FormatInt(int64(i), 10) + ")"
vendor/github.com/cilium/ebpf/asm/load_store_string.go:         return "Size(" + strconv.FormatInt(int64(i), 10) + ")"
vendor/github.com/cilium/ebpf/asm/opcode_string.go:             return "Class(" + strconv.FormatInt(int64(i), 10) + ")"
vendor/github.com/cilium/ebpf/syscalls.go:      return strconv.FormatInt(fd.raw, 10)
`